README.md

```diff
@@ -114,7 +114,7 @@ command right away and reset the interval.
 	-api-timeout duration
 	  Client timeout per request (default 5s)
 	-api-url string
-	  API URL. Takes precedence over HC_API_URL environment variable (default "https://hc-ping.com")
+	  API URL. Takes precedence over HC_API_URL environment variable. API URL needs to end in `/ping` e.g. `"https://my.hc/ping"` (default "https://hc-ping.com") 
```

Comment:
Strictly speaking the additional information in here is not correct. It assumes a default install of Healthchecks where the owner didn't set a `PING_ENDPOINT` environment variable so [the default](https://github.com/healthchecks/healthchecks/blob/b062caa2eb6aae794ae808d6a2a545900b8eabaa/hc/settings.py#L190) takes over (`SITE_ROOT + "/ping/").

So, it really depends on the install of the self hosted instance.

I'm inclined towards not making this change. I'd rather go with the assumption of self hoster to be equipped with knowledge to figure out what this value needs to be either looking at their logs or reading the source, than possibly misleading the user to always append the HC's default URI path.

Reply:
That's very interesting. I had no idea about the `PING_ENDPOINT` environment variable. I'm running the LinuxServer.io Healthchecks container. They don't mention the variable in any of their docs, it must be set in the container by default. I'll do a bit more digging and report back.

Reply:
A small clarification on the `PING_ENDPOINT` setting: it is used in the UI and in the documentation to format ping URLs. Changing ping URL by itself does not change how Healthchecks routes ping requests, it only changes how ping URLs are constructed for display in the UI and docs. 

Site operator can configure their web server to listen for pings on a different path (say, 'my.hc/p/` instead of `my.hc/ping/`), on a different subdomain (`ping.my.hc/`) or on a separate domain. In that case they can use the PING_ENDPOINT setting to "fix" the ping URLs displayed in the UI.

Reply:
I can't see the setting in my self-hosted UI, can you point me to it? As far as I can tell, my install is pretty default, I certainly didn't set up a different ping path, I'm using the default config from the SWAG template: https://github.com/linuxserver/reverse-proxy-confs/blob/master/healthchecks.subdomain.conf.sample

I completely agree with you though, if this is the standard config, and it's mine that's out of whack, the documentation should remain unchanged.

Reply:
@revilo951 PING_ENDPOINT is a configuration setting, it is initialized in [`settings.py`](https://github.com/healthchecks/healthchecks/blob/b062caa2eb6aae794ae808d6a2a545900b8eabaa/hc/settings.py#L190) from `PING_ENDPOINT` env variable, and it can be overriden in `local_settings.py`.

Perhaps the -api-url description could contain a note saying the `/ping` part *may* be needed, and give an example:

> -api-url string
> API URL. Takes precedence over HC_API_URL environment variable (default "https://hc-ping.com"). Must not contain a trailing slash. When using with a self-hosted Healthchecks instance, where ping endpoints are hosted under the "/ping" path, the API URL must include the path. Example for a Healthchecks instance running on https://example.org: `-api-url=https://example.org/ping`.

Reply:
I must have misread you before, it makes sense in hindsight. 

I like what you've suggested - hopefully it'll save someone else half a day of going around in circles. 😄